lib/AST/UnqualifiedLookup.cpp

```diff
@@ -605,25 +605,21 @@ bool ASTScopeDeclConsumerForUnqualifiedLookup::consume(
 
     auto fullName = factory.Name.getFullName();
     if (!value->getName().matchesRef(fullName)) {
-      if (!factory.options.contains(UnqualifiedLookupFlags::IncludePropertyWrapperResults))
-        continue;
-
-      auto *varDecl = dyn_cast<VarDecl>(value);
-      if (!varDecl || !varDecl->hasAttachedPropertyWrapper())
-        continue;
-
-      auto wrapperInfo = varDecl->getPropertyWrapperBackingPropertyInfo();
-      if (!wrapperInfo)
-        continue;
+      bool foundMatch = false;
+      if (auto *varDecl = dyn_cast<VarDecl>(value)) {
+        // Check if the name matches any auxiliary decls in the same scope
+        // that are not in the AST.
+        varDecl->visitAuxiliaryDecls([&](VarDecl *auxiliaryVar) {
+          if (auxiliaryVar->getDeclContext() == varDecl->getDeclContext() &&
```

Comment:
When is the DeclContext check expected to fail?

Reply:
Currently never, this is just a sanity check. I can remove it or make it an assertion